vibrato/src/dictionary/character.rs

```diff
@@ -127,4 +134,233 @@ impl CharProperty {
     pub fn num_categories(&self) -> usize {
         self.categories.len()
     }
+
+    /// Creates a new instance from `char.def`.
+    pub fn from_reader<R>(rdr: R) -> Result<Self>
+    where
+        R: Read,
+    {
+        let mut cate2info = BTreeMap::new();
```

Comment:
Why use `BTreeMap` instead of `HashMap`?

Reply:
cate2info and cate_map store information on character definitions, where the size is bounded by the number of character types (i.e., DEFAULT, SPACE, KANJI, and so on).

The scale is limited in practice, so there would be no advantage to employing hashing.

Reply:
It would be best if you considered using a collection with purposeful functions.

The following page describes the purpose of each collection.
https://doc.rust-lang.org/std/collections/index.html#use-a-hashmap-when
In this case, HashMap is the best choice.

Reply:
I see and agree with following the rust policy that says HashMap should be used in simple mapping cases.

On the one hand, this PR should focus only on removing builder modules. I'll do the replacement in another PR.